src/sickle.ts

```diff
@@ -399,6 +406,75 @@ class Annotator {
     }
   }
 
+  private visitExterns(node: ts.Node, namespace: string[] = []) {
+    this.currentOutput = this.externsOutput;
+    switch (node.kind) {
+      case ts.SyntaxKind.ModuleDeclaration:
+        let decl = <ts.ModuleDeclaration>node;
+        namespace = namespace.concat([decl.name.text]);
+        this.visitExterns(decl.body, namespace);
+        break;
+      case ts.SyntaxKind.ModuleBlock:
+        let block = <ts.ModuleBlock>node;
+        for (let stmt of block.statements) {
+          this.visitExterns(stmt, namespace);
+        }
+        break;
+      case ts.SyntaxKind.InterfaceDeclaration:
+        this.writeExternsInterface(<ts.InterfaceDeclaration>node, namespace);
+        break;
+      case ts.SyntaxKind.VariableStatement:
+        for (let decl of(<ts.VariableStatement>node).declarationList.declarations) {
+          this.writeExternsVariable(decl, namespace);
+        }
+        break;
+      default:
+        this.errorUnimplementedKind(node, 'externs generation');
+        break;
+    }
+    this.currentOutput = this.tsOutput;
+  }
+
+  private writeExternsInterface(decl: ts.InterfaceDeclaration, namespace: string[]) {
+    this.emit('/** @record @struct */\n');
```

Comment:
for my own understanding does @struct add anything on top of @record? I am used to seeing @struct on concrete classes to prevent adding fields post construction.